tests/test_decoding.py

```diff
@@ -499,8 +495,8 @@ def test_decode_nrc_const(self) -> None:
             sdgs=[],
         )
 
-        diag_layer_raw = DiagLayerRaw(
-            variant_type=DiagLayerType.BASE_VARIANT,
+        ecu_variant_raw = EcuVariantRaw(
+            variant_type=DiagLayerType.ECU_VARIANT,
```

Comment:
why?

Reply:
because ECU variants are the only kinds of diag layers which are supposed to directly describe of how to  communicate with a concrete ECU. (base variants are just helpers from which concrete ECU variants can be derived from)

Reply:
In practice, not quite, I have seen multiple cases where a Base variant is used to provide communication information for a whole group of ECUs

Usually it goes something like
* ECU variant is used for flashing, or to describe features specific to one ECU model.
* Base variant is used when describing features that are valid for multiple ECU models/variants

I have seen PDX files that have Base variants, but no ECU variants at all, and those are used to communicate with an entire group of cars, but provide only the generic shared diagnostic jobs

Reply:
yes, base variants describe common functionality for a range of ECU variants and can in principle also be used for communication, but if you have a piece of hardware in front of you, I'd strongly advise to use the ECU-Variant that targets the firmware version which is currently flashed on that hardware so that the quirks of that specific version can be covered. (I think it is plausible that some vendors do not clearly distinguish between base variants and ECU variants, but I think that `odxtools`'s unit tests should adhere to the intended way unless the test is explicitly about non-conformant behavior.)

Reply:
The idea is that the same API available to ECU variant, should also be available to Base variant

Reply:
Ah, I see: your point was primarily about test coverage? fixed (https://github.com/mercedes-benz/odxtools/pull/326/commits/5038cf69ce6ad7155b5e2f3a50d1d97177c3f42a), thanks!